File: docs/breez-sdk/snippets/python/src/messages.py
import logging
from breez_sdk_spark import BreezClient, CheckMessageRequest, SignMessageRequest


async def sign_message(client: BreezClient):
    # ANCHOR: sign-message
    message = "<message to sign>"
    # Set to true to get a compact signature rather than a DER
    compact = True
    try:
        sign_message_request = SignMessageRequest(
            message=message, compact=compact
        )
        sign_message_response = await client.message().sign(request=sign_message_request)

        signature = sign_message_response.signature
        pubkey = sign_message_response.pubkey

        logging.debug(f"Pubkey: {pubkey}")
        logging.debug(f"Signature: {signature}")
    except Exception as error:
        logging.error(error)
        raise
    # ANCHOR_END: sign-message


async def check_message(client: BreezClient):
    # ANCHOR: check-message
    message = "<message>"
    pubkey = "<pubkey of signer>"
    signature = "<message signature>"
    try:
        check_message_request = CheckMessageRequest(
            message=message, pubkey=pubkey, signature=signature
        )
        check_message_response = await client.message().check(request=check_message_request)

        is_valid = check_message_response.is_valid

        logging.debug(f"Signature valid: {is_valid}")
    except Exception as error:
        logging.error(error)
        raise
    # ANCHOR_END: check-message

File: docs/breez-sdk/snippets/csharp/Messages.cs
using Breez.Sdk.Spark;

namespace BreezSdkSnippets
{
    class Messages
    {
        async Task SignMessage(BreezClient client)
        {
            // ANCHOR: sign-message
            var message = "<message to sign>";
            // Set to true to get a compact signature rather than a DER
            var compact = true;
            var signMessageRequest = new SignMessageRequest(
                message: message,
                compact: compact
            );
            var signMessageResponse = await client.SignMessage(request: signMessageRequest);

            var signature = signMessageResponse.signature;
            var pubkey = signMessageResponse.pubkey;

            Console.WriteLine($"Pubkey: {pubkey}");
            Console.WriteLine($"Signature: {signature}");
            // ANCHOR_END: sign-message
        }

        async Task CheckMessage(BreezClient client)
        {
            // ANCHOR: check-message
            var message = "<message>";
            var pubkey = "<pubkey of signer>";
            var signature = "<message signature>";
            var checkMessageRequest = new CheckMessageRequest(
                message: message,
                pubkey: pubkey,
                signature: signature
            );
            var checkMessageResponse = await client.CheckMessage(request: checkMessageRequest);

            var isValid = checkMessageResponse.isValid;

            Console.WriteLine($"Signature valid: {isValid}");
            // ANCHOR_END: check-message
        }
    }
}
